### PR TITLE
Add OBO tests and minor changes to tests and dev apps

### PR DESCRIPTION
**Pull request by pmaytak**
Needs to be merged after #3048.

**Changes proposed in this request**
- Add more descriptive `LogLevel` comments.
- Minor perf test renamings.
- Add two OBO related integration tests.
- Update some dev apps to use cca config for lab.

**Testing**
N/A

**Performance impact**
N/A

### CODE REVIEW

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests.cs [hunk 1]
```diff
@@ -52,6 +53,180 @@ public void TestInitialize()
...
 
+        /// <summary>
+        /// Tests the behavior when calling OBO and silent in different orders with multiple users.
+        /// OBO calls should return tokens for correct users, silent calls should throw.
+        /// </summary>
+        [DataTestMethod]
+        [DataRow(false, false)]
+        [DataRow(true, false)]
+        [DataRow(true, true)]
+        public async Task OboAndSilent_Async(bool serializeCache, bool usePartitionedSerializationCache)
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests.cs:64]: nit / no action needed: integration tests are slow so we need to pick and choose scenarios where we use integration testing vs unit testing. Our unit testing strategy, where we fake out AAD, is pretty good (not perfect, needs more flexibility, but pretty good).

**Review by bgavrilMS — APPROVED** [review]

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests.cs (hunk 1 above)
Comment (pmaytak) [reply tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests.cs:64]: Good point. When I was investigating these, I think I went with integration tests to make sure the E2E is correct and I don't mock something incorectly.

**Comment by trwalke** [comment]
>     /// 1 - 100,000 - 100,000

Should probably update comment to reflect that these have changed

---
Refers to: tests/Microsoft.Identity.Test.Performance/AcquireTokenForOboCacheTests.cs:26 in 5db3247. [](commit_id = 5db3247fd6dd234a2ce55a2ffef19fbed9eb9ecf, deletion_comment = False)

**Comment by trwalke** [comment]
>     /// 1 - 100,000 - 100,000

Should probably update comment to reflect that these have changed

---
Refers to: tests/Microsoft.Identity.Test.Performance/TokenCacheTests.cs:25 in 5db3247. [](commit_id = 5db3247fd6dd234a2ce55a2ffef19fbed9eb9ecf, deletion_comment = False)

**Review by trwalke — APPROVED** [review]

**Review by gladjohn — APPROVED** [review]
looks good. Filed a task on lab to see if they can expose a different way to get users without having to hardcode UPNs.

**Pull request merged**